src/app/model/ChessEngine.ts

```diff
@@ -1,3 +1,48 @@
 import { ChessModel } from './ChessModel';
+import { Side, Square } from './Types';
+import { Piece } from './Piece';
+import { Player } from './Player';
+import { ChessBoard } from './ChessBoard';
+import { PieceWasMoved } from './PieceWasMoved';
+import { PieceWasCaptured } from './PieceWasCaptured';
 
-export class ChessEngine implements ChessModel {}
+export class ChessEngine implements ChessModel {
+  currentSide: Side = Side.BLACK;
+
+  constructor(readonly board: ChessBoard) {}
+
+  move(byPlayer: Player, squareFrom: Square, squareTo: Square): (PieceWasMoved | PieceWasCaptured)[] {
+    const chosenPiece = this.board.onPositionPiece(squareFrom) as Piece;
```

Comment:
Brakuje jednego przypadku.Powinienes sprawdzic czy to nie jest undefined :) czy ktos np. nie probuje ruszyc z pola na ktorym nie ma zadnego pionka.